lib/public/views/QcFlags/ForDataPass/QcFlagsForDataPassOverviewPage.js

```diff
@@ -12,22 +12,54 @@
  * or submit itself to any jurisdiction.
  */
 
-import { h } from '/js/src/index.js';
+import { h, iconWarning } from '/js/src/index.js';
 import { QcFlagsOverviewComponenet } from '../Overview/qcFlagOverviewComponent.js';
 import { frontLink } from '../../../components/common/navigation/frontLink.js';
+import { tooltip } from '../../../components/common/popover/tooltip.js';
+import { breadcrumbs } from '../../../components/common/navigation/breadcrumbs.js';
+import spinner from '../../../components/common/spinner.js';
 
 /**
  * Render Quality Control Flags For Data Pass Overview page
  * @param {Model} model The overall model object.
  * @returns {Component} The overview page
  */
 export const QcFlagsForDataPassOverviewPage = ({ qcFlags: { forDataPassOverviewModel: qcFlagsForDataPassOverviewModel } }) => {
-    /**
-     * Create links to runs per Data pass
-     * @param {DataPass} dataPass the Data pass
-     * @return {Componenet} in-breadcrumbs link to runs per Data page
-     */
-    const productionLinkFactory = (dataPass) =>
-        h('h2', frontLink(dataPass.name, 'runs-per-data-pass', { dataPassId: dataPass.id }));
-    return QcFlagsOverviewComponenet(qcFlagsForDataPassOverviewModel, { productionLinkFactory });
+    const {
+        run: remoteRun,
+        dplDetector: remoteDplDetector,
+        dataPass: remoteDataPass,
+    } = qcFlagsForDataPassOverviewModel;
+
+    const commonTitle = h('h2', 'QC');
+    return h('', {
+        onremove: () => qcFlagsForDataPassOverviewModel.reset(),
+    }, [
+        h('.flex-row.justify-between.items-center', [
+            h('.flex-row.g1.items-center', [
+                remoteDataPass.match({
+                    Success: (dataPass) => remoteRun.match({
+                        Success: (run) => remoteDplDetector.match({
+                            Success: (dplDetector) => breadcrumbs([
+                                commonTitle,
+                                h('h2', frontLink(dataPass.name, 'runs-per-data-pass', { dataPassId: dataPass.id })),
+                                h('h2', frontLink(run.runNumber, 'run-detail', { runNumber: run.runNumber })),
+                                h('h2', dplDetector.name),
+                            ]),
+                            Failure: () => [commonTitle, tooltip(h('.f3', iconWarning()), 'Not able to load detector info')],
+                            Loading: () => [commonTitle, h('', spinner({ size: 2, absolute: false }))],
+                            NotAsked: () => [commonTitle, tooltip(h('.f3', iconWarning()), 'No data was asked for')],
```

Comment:
When you see yourself writing data, it means that you need to be specific. What is data not asked for, runs, data pass... ?
Same in the other component

Reply:
applied